pulpcore/app/openapigenerator.py

```diff
@@ -459,18 +459,96 @@ def get_tags(self, operation_keys):
 
         return tags
 
-    def serializer_to_schema(self, serializer):
+    def get_default_response_serializer(self):
         """
-        Convert a serializer to an OpenAPI Schema.
-        Patch: https://github.com/axnsan12/drf-yasg/issues/70#issuecomment-485050813
+        Return the default response serializer for this endpoint.
+
+        Reference: https://github.com/axnsan12/drf-yasg/issues/70#issuecomment-485050813
+        """
+        body_override = self._get_request_body_override()
+        if body_override and body_override is not no_body:
+            return body_override
+
+        serializer = super().get_view_serializer()
+
+        if not serializer:
+            return
+
+        read_fields = OrderedDict()
+        for field_name, field in serializer.fields.items():
+            if not field.write_only:  # Removing write_only fields
+                read_fields[field_name] = field
+
+        serializer.fields = read_fields
+        return serializer
+
+    def get_view_serializer(self):
+        """
+        Return the appropriate serializer.
+
+        Reference: https://github.com/axnsan12/drf-yasg/issues/70#issuecomment-485050813
+        """
+
+        serializer = super().get_view_serializer()
+
+        if not serializer:
+            return
+
+        read_fields = OrderedDict()
+        write_fields = OrderedDict()
+        for field_name, field in serializer.fields.items():
+            if not field.write_only:  # Removing write_only fields
+                read_fields[field_name] = field
+            if not field.read_only:  # Removing read_only fields
+                write_fields[field_name] = field
+
+        if len(read_fields) == len(write_fields):
+            return serializer
+
+        contain_write_fields = len(serializer.fields) > len(read_fields)
+        if self.method.lower() == "get" or not contain_write_fields:
+            serializer.fields = read_fields
+            return serializer
```

Comment:
```suggestion
        contain_write_only_fields = len(serializer.fields) > len(read_fields)
        if self.method.lower() == "get" or not contain_write_only_fields:
            serializer.fields = read_fields
            return serializer
```
Also: are `serializer.fields` and `read_fields` not exactly the same in this case?